vaadin-checkbox-flow/src/main/java/com/vaadin/flow/component/checkbox/dataview/CheckboxGroupListDataViewImpl.java

```diff
@@ -0,0 +1,152 @@
+package com.vaadin.flow.component.checkbox.dataview;
+
+import com.vaadin.flow.component.checkbox.CheckboxGroup;
+import com.vaadin.flow.data.provider.*;
+import com.vaadin.flow.function.SerializableComparator;
+import com.vaadin.flow.function.SerializableConsumer;
+import com.vaadin.flow.function.SerializableFunction;
+import com.vaadin.flow.function.SerializablePredicate;
+import com.vaadin.flow.shared.Registration;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class CheckboxGroupListDataViewImpl<T> implements CheckboxGroupListDataView<T> {
+
+    private int itemsSize = 0;
+    private Set<SizeChangeListener> sizeChangeListeners;
+    private CheckboxGroup<T> component;
+
+    public CheckboxGroupListDataViewImpl(CheckboxGroup<T> checkboxGroup) {
+        this.component = checkboxGroup;
+    }
+
+    @Override
+    public T getNextItem(T item) {
+        return getNextOrPreviousItem(item, itemsList -> {
+            int itemIndex = itemsList.indexOf(item);
+            if (itemIndex != -1 && itemIndex < itemsList.size() - 1) {
+                return itemsList.get(itemIndex + 1);
+            }
+            return null;
+        });
+    }
+
+    @Override
+    public boolean hasNextItem(T item) {
+        return getNextItem(item) != null;
+    }
+
+    @Override
+    public T getPreviousItem(T item) {
+        return getNextOrPreviousItem(item, itemsList -> {
+            int itemIndex = itemsList.indexOf(item);
+            if (itemIndex > 0) {
+                return itemsList.get(itemIndex - 1);
+            }
+            return null;
+        });
+    }
+
+    @Override
+    public boolean hasPreviousItem(T item) {
+        return getPreviousItem(item) != null;
+    }
+
+    @Override
+    public CheckboxGroupListDataView<T> withFilter(SerializablePredicate<T> filter) {
+        CheckboxGroupListDataView<T> dataView = withFilterOrOrder(
+                dataProvider -> dataProvider.setFilter(filter));
+        // fire size change event if size has changed
+        getDataSize();
```

Comment:
`getDataSize()` should not fire events as it would be a bit unexpected to request for size and get size events at the same time.

Reply:
Ok, I will separate get size and size change event from each other.

Reply:
Done